.github/workflows/changelog-cd.yaml

```diff
@@ -0,0 +1,135 @@
+---
+name: Changelog generator CD
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - tools/changelog
+    tags:
+      - "tools/changelog/v[0-9]+.[0-9]+.[0-9]+**"
+  pull_request:
+    paths:
+      - tools/changelog/workflows/cd.yaml
+      - .github/workflows/changelog-cd.yaml
+
+env:
+  TOOL_DIRECTORY: tools/changelog
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # Needed to create the release
+      packages: write # Needed to upload the images to GHCR
+    steps:
+      # Setup
+      - name: Checkout repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - name: Create event-specific values
+        id: setup
+        run: |
+          # Determine if the workflow was triggered via a push to main or a tag
+          # and get the version based off of that
+          if [[ "${GITHUB_REF}" =~ refs/tags/.* ]]; then
+            # Transforms tag refs like refs/tags/tools/changelog/v1.2.3 into v1.2.3
+            echo "version=${GITHUB_REF#refs/tags/tools/changelog}" >> "${GITHUB_OUTPUT}"
+
+            # Eventually the parse-version action from the teleport.e repo
+            # should move into this repo and replace this logic
+            echo "should-release=true" >> "${GITHUB_OUTPUT}"
+            
+            if [[ "${GITHUB_REF%-*}" == "${GITHUB_REF}" ]]; then
```

Comment:
This test looks backwards to me - if you strip off a `-*` suffix and it is the same as itself. that means it did not have the suffix and is not a prerelease.

This could more simply just match against a glob: `[[ "${GITHUB_REF}" == *-* ]]` (that is safe inside `[[` - it won't match files or anything unwanted like that)